packages/block-library/src/button/style.scss

```
@@ -49,7 +49,11 @@ $blocks-button__height: 56px;
 
 .is-style-outline .wp-block-button__link,
 .wp-block-button__link.is-style-outline {
-	color: $dark-gray-700;
 	background-color: transparent;
 	border: 2px solid;
 }
+
+.is-style-outline .wp-block-button__link:not(.has-text-color),
```

To reduce potential of style conflicts:
```suggestion
.is-style-outline > .wp-block-button__link:not(.has-text-color),
```

(See also: #24919.)

---

Thanks. I've updated the file and applied your fix where needed.